### PR TITLE
Pull request for new version of SZ1.4.11

### PR DESCRIPTION
**Pull request by disheng222**
Hi,
I didn't realize the previous SPACK version I was working on is not the latest one. 
I have pulled the SPACK again, and there should be no merge conflicts now. 
Please use the latest version of SZ instead in your package. 
We added the executable (called sz) in the new version, and also fixed some bugs. 

Thanks.

Best,
Sheng

### CODE REVIEW

File: var/spack/repos/builtin/packages/sz/package.py [hunk 1]
```diff
@@ -30,11 +30,11 @@ class Sz(AutotoolsPackage):
     """Error-bounded Lossy Compressor for HPC Data."""
 
     homepage = "https://collab.cels.anl.gov/display/ESR/SZ"
-    url      = "https://github.com/disheng222/SZ/archive/v1.4.9.2.tar.gz"
+    url      = "https://github.com/disheng222/SZ/archive/v1.4.11.0.tar.gz"
 
     version('develop', git='https://github.com/disheng222/SZ.git',
             branch='master')
-    version('1.4.9.2', '028ce90165b7a4c4051d4c0189f193c0')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/sz/package.py]: Keep the old version!

**Comment by disheng222** [comment]
OK. I see.
You mean, when I perform the "spack create", I need to choose all versions
instead of only the latest version, right?
For instance:
==> Found 4 versions of sz:

  1.4.11.0  https://github.com/disheng222/SZ/archive/v1.4.11.0.tar.gz
  1.4.10.0  https://github.com/disheng222/SZ/archive/v1.4.10.0.tar.gz
  1.4.9.2   https://github.com/disheng222/SZ/archive/v1.4.9.2.tar.gz
  1.4.9     https://github.com/disheng222/SZ/archive/v1.4.9.tar.gz

==> How many would you like to checksum? (default is 1, q to abort) 4
........

Then, the package.py will look like this:

    url      = "https://github.com/disheng222/SZ/archive/v1.4.11.0.tar.gz"

    version('1.4.11.0', '10dee28b3503821579ce35a50e352cc6')
    version('1.4.10.0', '82e23dc5a51bcce1f70ba7e3b68a5965')
    version('1.4.9.2',  '028ce90165b7a4c4051d4c0189f193c0')
    version('1.4.9',    '10e861bad170c597fc730e290e311de4')


Best,
Sheng


2017-09-27 15:53 GMT-05:00 Denis Davydov <notifications@github.com>:

**Comment by davydden** [comment]
> You mean, when I perform the "spack create", I need to choose all versions

not really.

Say today you add `foo@1.2.3`. I use it to build something else in Spack and produce some results. It is really good to be able to reproduce the same build half year later, even when you have already `foo@5.6.7` around with bugfixes and new features. 

If today you have a couple of versions available to add, i.e. `1.2.3` and `1.2.4`. You don't have to add all of them. But once you add something, it's good to keep it around. That's why you see that other contributors request to keep your `1.4.9.2` around alongside the new release.

**Comment by disheng222** [comment]
Hi Denis,
You are right. The old version should be kept available anyway.
May I know how to keep it around for users? I mean, how to just add a new
version?
It seems that I just need to add one more item in package.py by using some
spack command or so, right?

I am a newbie to SPACK. Sorry for the bothering you so much...
Thanks.

Best,
Sheng

2017-09-27 16:03 GMT-05:00 Denis Davydov <notifications@github.com>:

**Comment by davydden** [comment]
> May I know how to keep it around for users? I mean, how to just add a new
version?

just don't delete  this line:

> version('1.4.9.2', '028ce90165b7a4c4051d4c0189f193c0')

I suggest reading http://spack.readthedocs.io/en/latest/contribution_guide.html and some links there to get familiar with Git.

**Comment by disheng222** [comment]
2017-09-27 16:19 GMT-05:00 Denis Davydov <notifications@github.com>:

> May I know how to keep it around for users? I mean, how to just add a new
> version?
>
> just don't delete this line:
>
> version('1.4.9.2', '028ce90165b7a4c4051d4c0189f193c0')
>
> Got it!

> I suggest reading http://spack.readthedocs.io/
> en/latest/contribution_guide.html and some links there to get familiar
> with Git.
>
Thanks.

**Comment by disheng222** [comment]
Hi,
I added older versions of SZ back into the package.py, and please approve
it.
Thanks.

Best,
Sheng

2017-09-27 16:26 GMT-05:00 Denis Davydov <notifications@github.com>:

**Comment by scheibelp** [comment]
(since the request made by the rejected review was satisfied, I merged the PR)

**Pull request merged**